src/Upload.js

```diff
@@ -0,0 +1,180 @@
+import '@reshuffle/code-transform/macro';
+import React, { useState, useEffect, useLayoutEffect, useCallback } from 'react';
+import { useDropzone } from 'react-dropzone';
+import clsx from 'clsx';
+import ReactCrop from 'react-image-crop';
+import { useFileUpload } from '@reshuffle/react-storage';
+import 'react-image-crop/dist/ReactCrop.css';
+import './Upload.css';
+import { setProfilePic } from '../backend/users';
+
+/**
+ * Calculate margin (m) between container (C) and centered object (O)
+ *
+ * CCCCCCCCCCCC
+ * mmmmOOOOmmmm
+ */
+function calcMargin(containerLength, objectLength) {
+  return (containerLength - objectLength) / 2;
+}
+
+/**
+ * Calculate object (O) crop (C) without given margin (m) => (D)
```

Comment:
Great, *much* better now!

Please use another letter for "crop" (`X` maybe), `C` is already "container" above.